Source/Core/VideoBackends/D3D12/StaticShaderCache.cpp

```diff
@@ -0,0 +1,699 @@
+	}
+
+	return bytecode;
+}
+
+D3D12_SHADER_BYTECODE StaticShaderCache::GetClearPixelShader()
+{
+	D3D12_SHADER_BYTECODE shader = {};
+	shader.BytecodeLength = s_clear_program_blob->Size();
+	shader.pShaderBytecode = s_clear_program_blob->Data();
+
+	return shader;
+}
+
+D3D12_SHADER_BYTECODE StaticShaderCache::GetAnaglyphPixelShader()
+{
+	D3D12_SHADER_BYTECODE shader = {};
+	shader.BytecodeLength = s_anaglyph_program_blob->Size();
+	shader.pShaderBytecode = s_anaglyph_program_blob->Data();
+
+	return shader;
+}
+
+D3D12_SHADER_BYTECODE StaticShaderCache::GetSimpleVertexShader()
+{
+	D3D12_SHADER_BYTECODE shader = {};
+	shader.BytecodeLength = s_simple_vertex_shader_blob->Size();
+	shader.pShaderBytecode = s_simple_vertex_shader_blob->Data();
+
+	return shader;
+}
+
+D3D12_SHADER_BYTECODE StaticShaderCache::GetClearVertexShader()
+{
+	D3D12_SHADER_BYTECODE shader = {};
+	shader.BytecodeLength = s_simple_clear_vertex_shader_blob->Size();
+	shader.pShaderBytecode = s_simple_clear_vertex_shader_blob->Data();
+
+	return shader;
+}
+
+D3D12_INPUT_LAYOUT_DESC StaticShaderCache::GetSimpleVertexShaderInputLayout()
+{
+	return s_simple_vertex_shader_input_layout;
+}
+
+D3D12_INPUT_LAYOUT_DESC StaticShaderCache::GetClearVertexShaderInputLayout()
+{
+	return s_clear_vertex_shader_input_layout;
+}
+
+D3D12_SHADER_BYTECODE StaticShaderCache::GetClearGeometryShader()
+{
+	D3D12_SHADER_BYTECODE bytecode = {};
+	if (g_ActiveConfig.iStereoMode > 0)
+	{
+		bytecode.BytecodeLength = s_clear_geometry_shader_blob->Size();
+		bytecode.pShaderBytecode = s_clear_geometry_shader_blob->Data();
+	}
+
+	return bytecode;
+}
+
+D3D12_SHADER_BYTECODE StaticShaderCache::GetCopyGeometryShader()
+{
+	D3D12_SHADER_BYTECODE bytecode = {};
+	if (g_ActiveConfig.iStereoMode > 0)
+	{
+		bytecode.BytecodeLength = s_copy_geometry_shader_blob->Size();
+		bytecode.pShaderBytecode = s_copy_geometry_shader_blob->Data();
+	}
+
+	return bytecode;
+}
+
+void StaticShaderCache::Init()
+{
+	// Compile static pixel shaders
+	D3D::CompilePixelShader(s_clear_program_hlsl, &s_clear_program_blob);
+	D3D::CompilePixelShader(s_anaglyph_program_hlsl, &s_anaglyph_program_blob);
+	D3D::CompilePixelShader(s_color_copy_program_hlsl, &s_color_copy_program_blob[0]);
+	D3D::CompilePixelShader(s_depth_copy_program_hlsl, &s_depth_copy_program_blob[0]);
+	D3D::CompilePixelShader(s_color_matrix_program_hlsl, &s_color_matrix_program_blob[0]);
+	D3D::CompilePixelShader(s_depth_matrix_program_hlsl, &s_depth_matrix_program_blob[0]);
+
+	// Compile static vertex shaders
+	D3D::CompileVertexShader(s_simple_vertex_shader_hlsl, &s_simple_vertex_shader_blob);
+	D3D::CompileVertexShader(s_clear_vertex_shader_hlsl, &s_simple_clear_vertex_shader_blob);
+
+	// Compile static geometry shaders
+	D3D::CompileGeometryShader(s_clear_geometry_shader_hlsl, &s_clear_geometry_shader_blob);
+	D3D::CompileGeometryShader(s_copy_geometry_shader_hlsl, &s_copy_geometry_shader_blob);
```

Comment:
Yeah, I knew there was something that could be better here.. does one map per shader type, with an enum as the key sound good, or something else?

For what it's worth, I also thought about putting these shaders in a disk cache, but the shaders are so small it only takes 33ms on startup to compile all of them.